lib/action_subscriber/rabbit_connection.rb

```diff
@@ -5,46 +5,27 @@ module RabbitConnection
     SUBSCRIBER_CONNECTION_MUTEX = ::Mutex.new
     NETWORK_RECOVERY_INTERVAL = 1.freeze
 
-    def self.connection_threadpools
-      if ::RUBY_PLATFORM == "java"
-        subscriber_connections.each_with_object({}) do |(name, connection), hash|
-          hash[name] = connection.instance_variable_get("@executor")
-        end
-      else
-        [] # TODO can I get a hold of the thredpool that bunny uses?
-      end
-    end
-
-    def self.setup_connection(name, settings)
-      SUBSCRIBER_CONNECTION_MUTEX.synchronize do
-        fail ArgumentError, "a #{name} connection already exists" if subscriber_connections[name]
-        subscriber_connections[name] = create_connection(settings)
-      end
-    end
-
     def self.subscriber_connected?
-      SUBSCRIBER_CONNECTION_MUTEX.synchronize do
-        subscriber_connections.all?{|_name, connection| connection.connected?}
-      end
+      with_connection{|connection| connection.connected? }
     end
 
     def self.subscriber_disconnect!
       SUBSCRIBER_CONNECTION_MUTEX.synchronize do
-        subscriber_connections.each{|_name, connection| connection.close}
-        @subscriber_connections = {}
+        @subscriber_connection.close if @subscriber_connection
+        @subscriber_connection = nil
       end
     end
 
-    def self.with_connection(name)
+    def self.with_connection
       SUBSCRIBER_CONNECTION_MUTEX.synchronize do
-        fail ArgumentError, "there is no connection named #{name}" unless subscriber_connections[name]
-        yield(subscriber_connections[name])
+        @subscriber_connection ||= create_connection
```

Comment:
Would we want to do something here like:
```
if @subscriber_connection &&  !@subscriber_connection.connected?
  @subscriber_connection.close # if this exists
  @subscriber_connection = nil
end

@subscriber_connection ||= create_connection
```
Or should we just wait for march_hare to reconnect?

Reply:
Right now we are relying on the march_hare/bunny reconnect logic so I think it makes sense to keep that behavior the same